pkgs/dart_services/lib/src/common_server.dart

```diff
@@ -202,6 +202,36 @@ class CommonServerApi {
     return ok(version().toJson());
   }
 
+  @Route.post('$apiPrefix/openInIDX')
+  Future<Response> openInIdx(Request request, String apiVersion) async {
+    final code = api.OpenInIdxRequest.fromJson(await request.readAsJson()).code;
+    final idxUrl = Uri.parse('https://idx.google.com/run.api');
+
+    final data = {
+      'project[files][lib/main.dart]': code,
+      'project[settings]': '{"baselineEnvironment": "flutter"}',
+    };
+    try {
+      final response = await http.post(
+        idxUrl,
+        body: data,
```

Comment:
It's a little odd to me that we're putting a map into the `body` field and not some encoded (json? urlencoded?) string.

Reply:
This is correct. From the documentation for `http.post`:

```
/// If [body] is a Map, it's encoded as form fields using [encoding]. The
/// content-type of the request will be set to
/// `"application/x-www-form-urlencoded"`; this cannot be overridden.
```